### DISCUSSION TITLE
why etuples if there are 'facts'?

### DISCUSSION THREAD

**Discussion by majidaldo** [pythological/kanren#80] [General]
etuples are designed to be 'programs' while the fact relations are 'data'. there's is also itero. (why not just go all in on lisp principles?)

is there a fundamental reason why these are distinct? what could you do with one that you can't with another?

**Comment by brandonwillard** [comment]
`etuple`s are just another way to represent Python computations in a "delayed", expressions-only way that works with `cons` semantics.  The caching is also pretty important.
